blocks/library/paragraph/index.js

```diff
@@ -265,20 +266,34 @@ registerBlockType( 'core/paragraph', {
 
 	save( { attributes } ) {
 		const { width, align, content, dropCap, backgroundColor, textColor, fontSize } = attributes;
+		const styles = {
+			backgroundColor: backgroundColor,
+			color: textColor,
+			fontSize: fontSize && fontSize + 'px',
+			textAlign: align,
+		};
+		const customClass = generateClass( styles, 'custom-paragraph-' );
 		const className = classnames( {
 			[ `align${ width }` ]: width,
 			'has-background': backgroundColor,
 			'has-drop-cap': dropCap,
+			[ customClass ]: customClass,
 		} );
+
+		return <p className={ className ? className : undefined }>{ content }</p>;
+	},
+
+	saveStyles( { attributes } ) {
```

Comment:
What if the user picks `white` as color and `black` as background-color for both button and paragraph? Shouldn't we share the same class names for both blocks? With the current implementation, we will create one combined class for all properties.

Reply:
We can also use in the paragraph the same approach as in the button. So classes can be reused in different block types, but for now, I intentionally used different approaches so we can see what is possible. The plan for colors from the theme and default Gutenberg palette is to use a set of classes this will only be used for colors outside the ones in the palette :)